### ISSUE TITLE
CA corrections desaturate image

### ISSUE THREAD

**Issue by gadolf66** [darktable-org/darktable#5604]
**Describe the bug**
After applying CA Corrections, image gets desaturated

**To Reproduce**

1. Get the raw file and xmp sidecar at [https://discuss.pixls.us/t/chromatic-aberrations-issue/18873/5?u=gadolf](https://discuss.pixls.us/t/chromatic-aberrations-issue/18873/5?u=gadolf)
2. Open the raw file and apply xmp file
3. Turn off CA Corrections
4. Take a snapshot
5. Turn on CA Corrections
6. Compare with snapshot

**Expected behavior**

Red and gree/yellow areas get desaturated when CA Corrections on

**Screenshots**

**Platform (please complete the following information):**
darktable 3.1.0~git2163.fda1e5796
opencl off
ubuntu 20.04

**Additional context**

I shared the xmp to apply dynamic range compression to this image, in order to ease showing the issue, since this image results from an ettr shot, thus, very dark.
Other user confirmed the issue on other images: [https://discuss.pixls.us/t/chromatic-aberrations-issue/18873/6?u=gadolf](https://discuss.pixls.us/t/chromatic-aberrations-issue/18873/6?u=gadolf)

**Comment by heckflosse** [comment]
> @heckflosse shared the code where he fixed the same issue in Rawtherapee, here

Just for information: The issue is not completely fixed in RawTherapee. There are very rare cases, where the fix in fact introduces a colour shift. These are for example images with extreme purple fringe.
Though, in my experience, it fixes the issue for 99.9 % of the images.

**Comment by jade-nl** [comment]
I can confirm this issue and it is not 3.1.0 specific, it also shows up in 3.0.2.

I can provide another RAW file (NEF, from a D750) that shows this issue if needed.

**Platform**
* darktable        : 3.1.0+2350~g3f399bdf8 + 3.0.2
* OpenCL installed : Yes (nvidia-opencl-common 440.82-1~bpo10+1)
* OpenCL activated : Yes
* Graphics card    : GeForce RTX 2060 SUPER7979 MiB
* Graphics driver  : nvidia (440.82)
* OS               : Linux (kernel 5.4.0-0.bpo.4-amd64)
* Distro           : Debian Buster (rel. 10.4)

**Comment by jade-nl** [comment]
I can confirm this issue and it is not 3.1.0 specific, it also shows up in 3.0.2.

I can provide another RAW file (NEF, from a D750) that shows this issue if needed.

**Platform**
* darktable        : 3.1.0+2350~g3f399bdf8 + 3.0.2
* OpenCL installed : Yes (nvidia-opencl-common 440.82-1~bpo10+1)
* OpenCL activated : Yes
* Graphics card    : GeForce RTX 2060 SUPER7979 MiB
* Graphics driver  : nvidia (440.82)
* OS               : Linux (kernel 5.4.0-0.bpo.4-amd64)
* Distro           : Debian Buster (rel. 10.4)

**Comment by jade-nl** [comment]
Just (re)checked: This is still the case in the latest darktable 3.1.0+2530 ge6d75ff00

**Comment by johnny-bit** [comment]
Yep, the issue is still present and needs to have fix from rawtherapee adapted to darktable.

**Comment by johnny-bit** [comment]
The fix is still needed... Need someone to port it ;(

**Comment by jenshannoschwalm** [comment]
Give me a few days....

**Comment by jenshannoschwalm** [comment]
progress here ...
